docs/operators/s3outputmodule.md

```diff
@@ -0,0 +1,199 @@
+    
+### Application Example
+-----------------------
+
+Please refer [Example](https://github.com/DataTorrent/examples/tree/master/tutorials/s3output) for S3OutputModule sample application.
+
+### Partitioning
+----------------
+
+Partitioning the module means that operator/operators in the module can able to be partitioned.
+
+#### Stateless Partitioning
+---------------------------
+
+Partitioning the operator in module can be achieved as follows:
+
+-   ***S3InitateFileUploadOperator***
+    -   Partition of this operator is not exposed from module. But, still can be achieved as follows:
+        ```xml
+        <property>
+          <name>dt.operator.{ModuleName}#InitiateUpload.attr.PARTITIONER</name>
+          <value>com.datatorrent.common.partitioner.StatelessPartitioner:{N}</value>
+        </property>     
+        ```
+
+        where {ModuleName} is the name of the S3OutputModule and
+              {N} is the number of static partitions.
+        Above lines will partition S3InitateFileUploadOperator statically {N} times. 
+
+-   ***S3BlockUploadOperator***
+    -   Locality of S3BlockUploadOperator with upstream operator must set to PARTITION_PARALLEL for performance benefits by avoiding serialization/deserialization of objects. So, Partitioning of this operator depends on upstream operator which is of type FSInputModule/BlockReader. 
+    
+-   ***S3FileMergerOperator***
+    -   By setting the parameter "mergerCount", S3FileMergerOperator be statically partitioned. This can be achieved by two ways:
+
+    -   (a) Following code can be added to populateDAG(DAG dag, Configuration conf) method of application to statically partitioning S3FileMergerOperator {N} times:
+
+        ```java
+        FSInputModule inputModule = dag.addModule("HDFSInputModule", new FSInputModule());
+        S3OutputTestModule outputModule = dag.addModule("S3OutputModule", new S3OutputTestModule());
+        outputModule.setMergerCount({N});
+        dag.addStream("FileMetaData", inputModule.filesMetadataOutput, outputModule.filesMetadataInput);
+        dag.addStream("BlocksMetaData", inputModule.blocksMetadataOutput, outputModule.blocksMetadataInput).setLocality(DAG.Locality.CONTAINER_LOCAL);
+        dag.addStream("BlocksData", inputModule.messages, outputModule.blockData).setLocality(DAG.Locality.CONTAINER_LOCAL);
+        ```
+
+    -   (b) By setting the parameter in properties file as follows
+
+        ```xml
+        <property>
+          <name>dt.operator.{ModuleName}.prop.mergerCount</name>
+          <value>{N}</value>
+        </property>     
+        ```
+
+        where {ModuleName} is the name of the S3OutputModule and
+              {N} is the number of static partitions.
+        Above lines will partition S3FileMergerOperator statically {N} times. 
+
+#### Dynamic Partitioning
+--------------------------
+
+Dynamic partitioning is a feature of Apex platform which changes the partition of the operator based on certain condition.
```

Comment:
the partition of the operator => the number of partitions of an operator
based on certain condition => at runtime